CMakeLists.txt

```diff
@@ -1845,6 +1845,11 @@ elseif(APPLE)
     file(GLOB SDLMAIN_SOURCES ${SDL2_SOURCE_DIR}/src/main/uikit/*.c)
   endif()
 
+  if(IOS OR TVOS OR MACOSX OR DARWIN)
+    file(GLOB SDL_VIDEO_CAPTURE_SOURCES ${SDL2_SOURCE_DIR}/src/video/SDL_video_capture.m)
```

Comment:
For darwin, you _possibly_ should check CoreMedia.framework existence.
(It appears to be present in SDKs 10.7+, and 10.7+ sdk is needed for build,
so possibly moot..) But macOS 10.6 is still supported at runtime, so possibly
weak linkage to CoreMedia.framework (see CheckJoystickMFI in autotools)
and possible runtime checks in *.m source?

Reply:
Maybe something like this - completely untested, cmake side not touched.
Maybe wrong and/or incomplete, @slouken would know better..
```diff
diff --git a/src/video/SDL_video_capture.m b/src/video/SDL_video_capture.m
index 43c04f2..c1e76c4 100644
--- a/src/video/SDL_video_capture.m
+++ b/src/video/SDL_video_capture.m
@@ -197,6 +197,12 @@ InitDevice(_THIS)
 
     AVCaptureDeviceFormat *spec_format = nil;
 
+#ifdef __MACOSX__
+    if (floor(NSAppKitVersionNumber) < NSAppKitVersionNumber10_7) {
+        return -1;
+    }
+#endif
+
     device = get_device_by_name(_this->dev_name);
     if (!device) {
         goto error;
diff --git a/configure.ac b/configure.ac
index b18d214..2f25f15 100644
--- a/configure.ac
+++ b/configure.ac
@@ -2181,6 +2181,7 @@ CheckCOCOA()
                   , enable_video_cocoa=yes)
     if test x$enable_video = xyes -a x$enable_video_cocoa = xyes; then
         save_CFLAGS="$CFLAGS"
+        save_LDFLAGS="$LDFLAGS"
 dnl     Work around that we don't have Objective-C support in autoconf
         CFLAGS="$CFLAGS -x objective-c"
         AC_MSG_CHECKING(for Cocoa framework)
@@ -2195,6 +2196,8 @@ dnl     Work around that we don't have Objective-C support in autoconf
             SUMMARY_video="${SUMMARY_video} cocoa"
             have_video=yes
         fi
+        CFLAGS="$save_CFLAGS -x objective-c -fobjc-weak"
+        LDFLAGS="$LDFLAGS -Wl,-weak_framework,CoreMedia"
         AC_MSG_CHECKING(for CoreMedia framework)
         have_coremedia=no
         AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
@@ -2202,6 +2205,7 @@ dnl     Work around that we don't have Objective-C support in autoconf
           #import <CoreMedia/CoreMedia.h>
         ]],[])], [have_coremedia=yes],[])
         CFLAGS="$save_CFLAGS"
+        LDFLAGS="$save_LDFLAGS"
         if test x$have_coremedia = xyes; then
             SOURCES="$SOURCES $srcdir/src/video/SDL_video_capture.m"
         fi
@@ -4335,7 +4339,7 @@ dnl BeOS support removed after SDL 2.0.1. Haiku still works.  --ryan.
         EXTRA_LDFLAGS="$EXTRA_LDFLAGS -lobjc"
         EXTRA_LDFLAGS="$EXTRA_LDFLAGS -Wl,-framework,CoreVideo"
         if test x$have_coremedia = xyes; then
-           EXTRA_LDFLAGS="$EXTRA_LDFLAGS -Wl,-framework,CoreMedia"
+           EXTRA_LDFLAGS="$EXTRA_LDFLAGS -Wl,-weak_framework,CoreMedia"
         fi
         EXTRA_LDFLAGS="$EXTRA_LDFLAGS -Wl,-framework,Cocoa"
         EXTRA_LDFLAGS="$EXTRA_LDFLAGS -Wl,-framework,Carbon"
```